### PR TITLE
Update iOS App Clip instructions for Xcode 15 and iOS 16

### PR DESCRIPTION
**Pull request by jmagman**
_Description of what this PR is changing or adding, and why:_
1. Update some Xcode screenshots to the current version of Xcode where the UI has significantly changed.
2. Mention the iOS 16 15MB app clip size limit.
3. Remove "Framework Search Paths" section, that was never necessary.

_Issues fixed by this PR (if any):_
See https://github.com/flutter/flutter/issues/71098#issuecomment-1941419920

https://flutter-docs-prod--pr10145-app-clip-jej14gwc.web.app/platform-integration/ios/ios-app-clip

## Presubmit checklist

- [x] This PR doesn’t contain automatically generated corrections (Grammarly or similar).
- [x] This PR follows the [Google Developer Documentation Style Guidelines](https://developers.google.com/style) — for example, it doesn’t use _i.e._ or _e.g._, and it avoids _I_ and _we_ (first person).
- [x] This PR uses [semantic line breaks](https://github.com/dart-lang/site-shared/blob/main/doc/writing-for-dart-and-flutter-websites.md#semantic-line-breaks) of 80 characters or fewer.

### CODE REVIEW

File: src/platform-integration/ios/ios-app-clip.md [hunk 1]
```diff
@@ -55,8 +54,6 @@ Enter your new target detail in the dialog.
 
 Select **Storyboard** for Interface.
 
-Select **UIKit App Delegate** for Life Cycle.
```
Comment (jmagman) [inline src/platform-integration/ios/ios-app-clip.md:58]: This field is no longer present in latest Xcode. I updated the screenshot.

File: src/platform-integration/ios/ios-app-clip.md [hunk 2]
```diff
@@ -74,6 +71,14 @@ activate the new scheme for the new target.
 {% include docs/app-figure.md
 image="development/platform-integration/ios-app-clip/activate-scheme.png" %}
 
+**2.5**
+
+Back in the project settings, open the **Build Phases** tab.
+Drag **Embedded App Clips** to above **Thin Binary**.
```
Comment (jmagman) [inline src/platform-integration/ios/ios-app-clip.md:77]: This avoids an error when you build.

File: src/platform-integration/ios/ios-app-clip.md [hunk 3]
```diff
@@ -223,22 +231,6 @@ These steps are not necessary for add-to-app.
 
 **7.1**
 
-In your App Clip's target's project settings,
```
Comment (jmagman) [inline src/platform-integration/ios/ios-app-clip.md:226]: This section was never necessary.

File: src/platform-integration/ios/ios-app-clip.md [hunk 4]
```diff
@@ -267,6 +259,8 @@ Expand the new phase and add this line to the script content:
 /bin/sh "$FLUTTER_ROOT/packages/flutter_tools/bin/xcode_backend.sh" build
 ```
 
+Uncheck **Based on dependency analysis**.
```
Comment (jmagman) [inline src/platform-integration/ios/ios-app-clip.md:262]: This is new and required to avoid a warning.

**Review by sfshaza2 — APPROVED** [review]
lgtm! Thanks, @jmagman!

**Comment by hellohuanlin** [comment]
curious how did you create the image with blur effect?

**Pull request merged**